query/regexp.go

```diff
@@ -42,3 +42,60 @@ func LowerRegexp(r *syntax.Regexp) *syntax.Regexp {
 
 	return &newRE
 }
+
+// optimizeRegexp converts capturing groups to non-capturing groups.
+// Returns original input if an error is encountered
+func optimizeRegexp(re *syntax.Regexp) *syntax.Regexp {
+	r := convertCapture(re)
+	return r.Simplify()
+}
+
+func convertCapture(re *syntax.Regexp) *syntax.Regexp {
+	if hasCapture(re) {
```

Comment:
style nit: you can avoid the extra indentation with an early return. eg `if !hasCapture { return re }`